### PR TITLE
site: Add Contour Ingress Controller as example in /docs/user/ingress

### PR DESCRIPTION
**Pull request by stevesloka**
This adds to the Ingress examples by showing how to deploy with Contour (projectcontour.io). 

Signed-off-by: Steve Sloka <steve@stevesloka.com>

### CODE REVIEW

**Comment by amwat** [comment]
https://deploy-preview-1247--k8s-kind.netlify.com/docs/user/ingress/#contour
/retest

File: site/content/docs/user/ingress.md [hunk 1]
```diff
@@ -25,8 +25,40 @@ by the ingress controller `nodeSelector`.
...
 The following ingress controllers are known to work:
 
+ - [Contour](#contour)
  - [Ingress NGINX](#ingress-nginx)
 
+### Contour
+
+Create a kind cluster with `extraPortMappings` which will expose ports 80 & 443 from your cluster to your local host.
+
+{{< codeFromInline lang="bash" >}}
```
Comment (amwat) [inline site/content/docs/user/ingress.md]: This part is similar to the ingress nginx portmappings? Can we merge them
You'll also want to use node labels for making it work in multi node clusters.

Comment (stevesloka) [reply site/content/docs/user/ingress.md]: The configs are similar but Contour doesn't require those extra mappings which is why I split the configs apart. That said, having them for Contour won't break it, just not needed to get it running with Kind. 

How would this work with multi-node clusters in kind? You can't have additional ports map to 80/443?

Comment (amwat) [reply site/content/docs/user/ingress.md]: What I meant is, in the multi-node case, those node-labels and the deployments patches are required to ensure that the controller gets scheduled correctly to the node which has the ports exposed.

For example I just tried running contour ingress against a cluster with some additional nodes and it doesn't work properly.

```
cat <<EOF | kind create cluster --config=-
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  extraPortMappings:
  - containerPort: 80
    hostPort: 80
    protocol: TCP
  - containerPort: 443
    hostPort: 443
    protocol: TCP
- role: control-plane
- role: worker
- role: worker
EOF
```

Comment (stevesloka) [reply site/content/docs/user/ingress.md]: Ahhh ok, I'm following now. =) Let me fix this up.

**Comment by stevesloka** [comment]
@amwat I did a little bit of re-org from your review comments. Please take a look. =)

File: site/content/docs/user/ingress.md [hunk 2]
```diff
@@ -55,6 +59,35 @@ nodes:
...
+```
+
+Apply it by running:
+
+{{< codeFromInline lang="bash" >}}
+kubectl patch daemonsets -n projectcontour envoy -p '{{< minify file="static/examples/ingress/contour/patch.json" >}}' 
+{{< /codeFromInline >}}
+
+Now the Contour is all setup to be used. 
+Refer the [Using Ingress](#using-ingress) for a basic example usage.
```
Comment (BenTheElder) [inline site/content/docs/user/ingress.md]: S/Refer the/Refer to/ ?

**Comment by amwat** [comment]
/lgtm
Thanks!

**Comment by BenTheElder** [comment]
/approve
@amwat mind fixing the minor typo when you file to fix the patching on older k8s as discussed in slack?

**Comment by BenTheElder** [comment]
Thanks @stevesloka !

**Pull request merged**